### PR TITLE
rtags: 2.3 -> 2.8

### PR DESCRIPTION
**Pull request by periklis**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Ralith** [comment]
You need to change `src.sha256` as well at minimum.

**Comment by periklis** [comment]
@Ralith Thx, i missed that.

File: pkgs/development/tools/rtags/default.nix
Comment (Ralith) [inline pkgs/development/tools/rtags/default.nix]: s/optional/optionals/

File: pkgs/development/tools/rtags/default.nix
Comment (Ralith) [inline pkgs/development/tools/rtags/default.nix]: Why was this added?

**Review by Ralith — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/rtags/default.nix
Comment (periklis) [inline pkgs/development/tools/rtags/default.nix]: According to the build process rtags needs exception support.

File: pkgs/development/tools/rtags/default.nix
Comment (periklis) [inline pkgs/development/tools/rtags/default.nix]: fixed

File: pkgs/development/tools/rtags/default.nix
Comment (Ralith) [inline pkgs/development/tools/rtags/default.nix]: I'm not sure what "according to the build process" means. There's multiple issues here: `-fexceptions` is enabled by default, overriding `NIX_CFLAGS_COMPILE` is the wrong way to specify cflags, and package configures scripts are in general responsible for choosing their compilation options themselves--it's unlikely that NixOS-specific customizations are needed outside of extending search paths.

Comment (Ralith) [inline pkgs/development/tools/rtags/default.nix]: On investigation, it looks like `-fno-exceptions` is coming from `$(llvm-config --cxxflags)` on line 11. If you add `-fexceptions` after that I'll approve these changes.

Comment (periklis) [inline pkgs/development/tools/rtags/default.nix]: Thx for the input, i am occasionally contributing to packages, so i am interrested to learn. You mean i should put FLAGS in CFLAGS or CXXFLAGS and not NIX_CFLAGS_COMPILE, right?

Comment (Ralith) [inline pkgs/development/tools/rtags/default.nix]: The exact method will depend on the build system used by the package in question. In this case, it seems like we needed to work around an issue with the flags we were already passing in a very rtags-specific environment variable.

**Review by Ralith — APPROVED** [review]
Looks good to me!

**Comment by periklis** [comment]
Hmm... something is wrong with the build of rdm.cpp. I am getting a strange compile error on OSX:
```
[ 86%] Building CXX object src/CMakeFiles/rdm.dir/rdm.cpp.o
In file included from /private/var/folders/tv/h06nsg4s31b9lcd0_wl1rcjh0000gn/T/nix-build-rtags-2.8.drv-0/rtags/src/rdm.cpp:32:
/private/var/folders/tv/h06nsg4s31b9lcd0_wl1rcjh0000gn/T/nix-build-rtags-2.8.drv-0/rtags/src/RTags.h:978:5: warning: default label in switch which covers all enumeration values [-Wc
overed-switch-default]                                                                                                                                                              
    default:
    ^
/private/var/folders/tv/h06nsg4s31b9lcd0_wl1rcjh0000gn/T/nix-build-rtags-2.8.drv-0/rtags/src/rdm.cpp:318:11: error: use of undeclared identifier 'FileManagerWatch'; did you mean 'No
FileManagerWatch'?                                                                                                                                                                  
        { FileManagerWatch, "filemanager-watch", 'M', CommandLineParser::NoValue, "Use a file system watcher for filemanager." },
          ^~~~~~~~~~~~~~~~
          NoFileManagerWatch
/private/var/folders/tv/h06nsg4s31b9lcd0_wl1rcjh0000gn/T/nix-build-rtags-2.8.drv-0/rtags/src/rdm.cpp:184:5: note: 'NoFileManagerWatch' declared here
    NoFileManagerWatch,
    ^
/private/var/folders/tv/h06nsg4s31b9lcd0_wl1rcjh0000gn/T/nix-build-rtags-2.8.drv-0/rtags/src/rdm.cpp:585:14: error: use of undeclared identifier 'FileManagerWatch'; did you mean 'No
FileManagerWatch'?                                                                                                                                                                  
        case FileManagerWatch: {
             ^~~~~~~~~~~~~~~~
             NoFileManagerWatch
/private/var/folders/tv/h06nsg4s31b9lcd0_wl1rcjh0000gn/T/nix-build-rtags-2.8.drv-0/rtags/src/rdm.cpp:184:5: note: 'NoFileManagerWatch' declared here
    NoFileManagerWatch,
    ^
1 warning and 2 errors generated.
```
Afaik the preprocessor conditionals on https://github.com/Andersbakken/rtags/blob/master/src/rdm.cpp#L39 and https://github.com/Andersbakken/rtags/blob/master/src/rdm.cpp#L181 are quite strange combination to me. Any idea?

**Comment by Ralith** [comment]
Looks like an upstream (i.e. rtags) bug at a glance; perhaps they're not testing releases on OSX. I don't have time to dig deeply into it right now, but you might try contacting the maintainer.

**Comment by periklis** [comment]
I've opened an issue on [Andersbakken/rtags #891](https://github.com/Andersbakken/rtags/issues/891)

**Comment by periklis** [comment]
@Mic92 If we merge, we will make our OSX users unhappy. If you don't bother i would like to wait for a response from upstream. Otherwise i would open an osx issue if that the way to go, what's your opinion?

**Comment by Mic92** [comment]
@periklis yes. If it breaks OS X, we should wait.

**Comment by periklis** [comment]
@Mic92 The rtags maintainer fixed the bug in upstream and i pinned the release to that commit. I've asked for a release tag and will deliver that asap. Now we can merge this PR.

**Pull request merged**
